test/Microsoft.DotNet.DarcLib.Tests/GitHubClientTests.cs

```diff
@@ -145,9 +155,31 @@ public void SetGitHubClientObject(IGitHubClient value)
     public override IGitHubClient GetClient(string repoUri) => _client;
 
     public TestGitHubClient(string gitExecutable, string accessToken, ILogger logger, string temporaryRepositoryPath, IMemoryCache cache)
-        : base(new ResolvedTokenProvider(accessToken), new ProcessManager(logger, gitExecutable), logger, temporaryRepositoryPath, cache)
+        : base(new ResolvedTokenProvider(accessToken), new ProcessManager(logger, gitExecutable), temporaryRepositoryPath, cache, new NoOpRedisClient(), logger)
     {
     }
+    protected override async Task<T> RequestResourceUsingEtagsAsync<T, K>(
+        string resourceKey,
+        Uri resourceUri,
+        IGitHubClient client,
+        Func<K, T> resourceConverter)
+    {
+        await Task.FromResult<T>(null);
```

Comment:
This `await` line should be deleted. The method does not need to be `async` and we can just return `Task.FromResult(...)`